src/elements/Label/Label.js

```diff
@@ -114,12 +131,6 @@ Label.propTypes = {
   /** Additional text with less emphasis. */
   detail: PropTypes.string,
 
-  /** An element type to render the 'detail' as (string or function). */
-  detailAs: PropTypes.oneOfType([
```

Comment:
Initially, I was going to suggest we make it a shorthand prop but the current method follows what was done for the [Dropdown.Header shorthand](https://github.com/TechnologyAdvice/stardust/pull/501/files).  Let's leave it as is and we'll update it later.

I'll try to focus on creating a generic shorthand factory right now. If it is memoized, we might be able to use it all over without creating multiple individual factories in `factories/index.js`.